src/libs/ReportUtils.ts

```diff
@@ -23,171 +26,193 @@ import isReportMessageAttachment from './isReportMessageAttachment';
+import {Beta, Login, PersonalDetails, Policy, Report, ReportAction, Transaction} from '../types/onyx';
+import {Receipt} from '../types/onyx/Transaction';
+import DeepValueOf from '../types/utils/DeepValueOf';
+import {Closed, IOUMessage} from '../types/onyx/OriginalMessage';
+import {Message, ReportActions} from '../types/onyx/ReportAction';
+
+type WelcomeMessage = {showReportName: boolean; phrase1?: string; phrase2?: string};
+
+type Avatar = {
+    id: number;
+    source: React.FC<SvgProps> | string | undefined;
+    type: typeof CONST.ICON_TYPE_AVATAR | typeof CONST.ICON_TYPE_WORKSPACE;
+    name: string;
+    fallbackIcon?: React.FC<SvgProps> | string;
+};
+
+type ExpanseOriginalMessage = {
+    oldComment?: string;
+    newComment?: string;
+    comment?: string;
+    merchant?: string;
+    oldCreated?: string;
+    created?: string;
+    oldMerchant?: string;
+    oldAmount?: number;
+    amount?: number;
+    oldCurrency?: string;
+    currency?: string;
+    category?: string;
+    oldCategory?: string;
+    tag?: string;
+    oldTag?: string;
+    billable?: string;
+    oldBillable?: string;
+};
 
-let currentUserEmail;
-let currentUserAccountID;
-let isAnonymousUser;
+type Participant = {
+    accountID: number;
+    alternateText: string;
+    firstName: string;
+    icons: Avatar[];
+    keyForList: string;
+    lastName: string;
+    login: string;
+    phoneNumber: string;
+    searchText: string;
+    selected: boolean;
+    text: string;
+};
+
+function isTypeTransaction(arg: Transaction | Record<string, never>): arg is Transaction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+function isTypeReportAction(arg: ReportAction | Record<string, never>): arg is ReportAction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+let currentUserEmail: string | undefined;
+let currentUserAccountID: number | undefined;
+let isAnonymousUser = false;
 
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
+    callback: (value) => {
         // When signed out, val is undefined
-        if (!val) {
+        if (!value) {
             return;
         }
 
-        currentUserEmail = val.email;
-        currentUserAccountID = val.accountID;
-        isAnonymousUser = val.authTokenType === 'anonymousAccount';
+        currentUserEmail = value.email;
+        currentUserAccountID = value.accountID;
+        isAnonymousUser = value.authTokenType === 'anonymousAccount';
     },
 });
 
-let allPersonalDetails;
-let currentUserPersonalDetails;
+let allPersonalDetails: OnyxCollection<PersonalDetails>;
+let currentUserPersonalDetails: OnyxEntry<PersonalDetails | undefined>;
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-    callback: (val) => {
-        currentUserPersonalDetails = lodashGet(val, currentUserAccountID, {});
-        allPersonalDetails = val || {};
+    callback: (value) => {
+        currentUserPersonalDetails = value?.[currentUserAccountID ?? ''];
```

Comment:
@kubabutkiewicz Before the default value was `{}`, make sure your update doesn't cause problems. Also you can do:
`currentUserPersonalDetails = value?.[currentUserAccountID] ?? null;`  to get rid of undefined in 106 line

Reply:
I think I cannot do that because `currentUserAccount` could be `undefined` and undefined cannot be used as index